### ISSUE TITLE
How to do a scaffold split and return smiles instead of features

### ISSUE THREAD

**Issue by aredze** [deepchem/deepchem#2455]
## ❓ Questions & Help

I want to use DeepChem to scaffold-split my input_dataset and return a train_dataset and test_dataset. 

My input_dataset has compound_ids (unique identifiers) and smiles (plus y's) and I would like my train_dataset and test_dataset also to have compound_ids and smiles (plus y's). 

I have tested the following lines of code.
`import deepchem as dc
featurizer = dc.feat.CircularFingerprint(size=1024)

comp_ids1 = ['mol1','mol2', 'mol3','mol4']
smiles1 = ['C', 'CC', 'CCCC', 'CCCC'] 

ecfp1 = featurizer.featurize(smiles)
ecfp2 = [1,1,1,1]

dataset1 = dc.data.NumpyDataset(X=ecfp1, y=[1,0,1,0], ids=smiles1)
display(dataset1.to_dataframe())

#splitter = dc.splits.RandomSplitter() 
splitter = dc.splits.ScaffoldSplitter() 

train_dataset, test_dataset = splitter.train_test_split( dataset=dataset1, frac_train=0.5, seed=0)

display(train_dataset.to_dataframe())
display(test_dataset.to_dataframe())`

1) How can I pass both comp_ids and smiles to the dataset so that I can retrieve both? 
2) Is there a reason why scaffold split requires the smiles to be the actual ids of the compounds?
3) How can I bypass the featurization step since I am not really using it in this case?
4) Finally, if I replace X=ecfp1 with X=ecfp2 in the NumpyDataset creation I get the wrong number of compounds in each dataset (potentially a bug).

**Comment by rbharath** [comment]
Excellent questions!

1. This is honestly a little awkward to do right now. One way I've done it is to maintain an auxiliary list of compound ids and have a dictionary mapping smiles to compounds ids so I can retrieve them later.
2. Backwards compatibility mainly! We might want to consider changing the design where but we'd have to be careful about deprecating the old design. Might be worth it if we can come up with a clean design
3. Check out `dc.feat.RawFeaturizer()`. It's a hack but designed for cases like yours where you have a no-op featurization. Alternatively, you can just construct a dataset "by hand" using the constructor
4. This might be due to a featurization failure? Can you put together a simple failing test case for us to look at?

**Comment by YojanaGadiya** [comment]
Has this issue been solved now @rbharath?
